### PR TITLE
4119 tomas

### PR DESCRIPTION
**Pull request by TomasLongo**

### CODE REVIEW

File: data-prepper-plugins/otel-metrics-source/README.md [hunk 1]
```diff
@@ -26,6 +26,18 @@ source:
     * `none`: no compression
     * `gzip`: apply GZip de-compression on the incoming request.
 
+### Retry Information
+
+Data Prepper gives clients a hint on how long to wait for the next request in case backpressure builds up. The retry information is implemented as exponential backoff, with a max delay of `retry_info.max_delay`.
```
Comment (KarstenSchnitter) [inline data-prepper-plugins/otel-metrics-source/README.md:31]: See comment in logs source README.md

File: data-prepper-plugins/otel-metrics-source/src/main/java/org/opensearch/dataprepper/plugins/source/otelmetrics/OTelMetricsSource.java [hunk 2]
```diff
@@ -224,6 +225,17 @@ public void stop() {
         LOG.info("Stopped otel_metrics_source.");
     }
 
+    private GrpcExceptionHandlerFunction createGrpExceptionHandler() {
+        Duration defaultMinDelay = Duration.ofMillis(100);
+        Duration defaultMaxDelay = Duration.ofMillis(2000);
```
Comment (KarstenSchnitter) [inline data-prepper-plugins/otel-metrics-source/src/main/java/org/opensearch/dataprepper/plugins/source/otelmetrics/OTelMetricsSource.java:230]: See comment in `OTelLogsSource`.

File: data-prepper-plugins/otel-metrics-source/src/test/java/org/opensearch/dataprepper/plugins/source/otelmetrics/OtelMetricsSourceConfigTests.java [hunk 3]
```diff
@@ -303,6 +304,29 @@ void testInValidConfigWithCustomPath() {
...
+                false,
+                false,
+                true,
+                TEST_KEY_CERT,
+                "",
+                DEFAULT_THREAD_COUNT,
+                DEFAULT_MAX_CONNECTION_COUNT);
+
+        final OTelMetricsSourceConfig otelTraceSourceConfig = OBJECT_MAPPER.convertValue(customPathPluginSetting.getSettings(), OTelMetricsSourceConfig.class);
+
```
Comment (KarstenSchnitter) [inline data-prepper-plugins/otel-metrics-source/src/test/java/org/opensearch/dataprepper/plugins/source/otelmetrics/OtelMetricsSourceConfigTests.java:323]:
```suggestion
```

File: data-prepper-plugins/otel-trace-source/README.md [hunk 4]
```diff
@@ -42,6 +42,19 @@ For more information on migrating from Data Prepper 1.x to Data Prepper 2.x, see
     * `none`: no compression
     * `gzip`: apply GZip de-compression on the incoming request.
 
+
+### Retry Information
+
+Data Prepper gives clients a hint on how long to wait for the next request in case backpressure builds up. The retry information is implemented as exponential backoff, with a max delay of `retry_info.max_delay`.
```
Comment (KarstenSchnitter) [inline data-prepper-plugins/otel-trace-source/README.md:48]: See comment in logs source README.md.

File: data-prepper-plugins/otel-trace-source/src/main/java/org/opensearch/dataprepper/plugins/source/oteltrace/OTelTraceSource.java [hunk 5]
```diff
@@ -208,6 +208,17 @@ public void start(Buffer<Record<Object>> buffer) {
         LOG.info("Started otel_trace_source on port " + oTelTraceSourceConfig.getPort() + "...");
     }
 
+    private GrpcExceptionHandlerFunction createGrpExceptionHandler() {
+        Duration defaultMinDelay = Duration.ofMillis(100);
+        Duration defaultMaxDelay = Duration.ofMillis(2000);
```
Comment (KarstenSchnitter) [inline data-prepper-plugins/otel-trace-source/src/main/java/org/opensearch/dataprepper/plugins/source/oteltrace/OTelTraceSource.java:213]: See comment in `OTelLogsSource`.

File: data-prepper-plugins/otel-logs-source/README.md [hunk 6]
```diff
@@ -26,6 +26,19 @@ source:
...
+
+Data Prepper gives clients a hint on how long to wait for the next request in case backpressure builds up. The retry information is implemented as exponential backoff, with a max delay of `retry_info.max_delay`.
+
+```yaml
+source:
+  otel_trace_source:
+    retry_info:
+      min_delay: 1000ms # defaults to 100ms
+      max_delay: 5s     # defaults to 2s
+
```
[the remark this answers is not in the document]
Comment (i-vim) [reply data-prepper-plugins/otel-logs-source/README.md:39]: done

File: data-prepper-plugins/otel-logs-source/README.md [hunk 7]
```diff
@@ -26,6 +26,19 @@ source:
     * `none`: no compression
     * `gzip`: apply GZip de-compression on the incoming request.
 
+### Retry Information
+
+Data Prepper gives clients a hint on how long to wait for the next request in case backpressure builds up. The retry information is implemented as exponential backoff, with a max delay of `retry_info.max_delay`.
```
[the remark this answers is not in the document]
Comment (i-vim) [reply data-prepper-plugins/otel-logs-source/README.md:31]: done

File: data-prepper-plugins/otel-logs-source/src/main/java/org/opensearch/dataprepper/plugins/source/otellogs/OTelLogsSource.java [hunk 8]
```diff
@@ -205,6 +206,17 @@ public void stop() {
         LOG.info("Stopped otel_logs_source.");
     }
 
+    private GrpcExceptionHandlerFunction createGrpExceptionHandler() {
+        Duration defaultMinDelay = Duration.ofMillis(100);
+        Duration defaultMaxDelay = Duration.ofMillis(2000);
```
[the remark this answers is not in the document]
Comment (i-vim) [reply data-prepper-plugins/otel-logs-source/src/main/java/org/opensearch/dataprepper/plugins/source/otellogs/OTelLogsSource.java:211]: done

File: data-prepper-plugins/otel-logs-source/src/test/java/org/opensearch/dataprepper/plugins/source/otellogs/OtelLogsSourceConfigTests.java [hunk 9]
```diff
@@ -299,6 +323,7 @@ private PluginSetting completePluginSettingForOtelLogsSource(final int requestTi
         settings.put(SSL_KEY_FILE, sslKeyFile);
         settings.put(THREAD_COUNT, threadCount);
         settings.put(MAX_CONNECTION_COUNT, maxConnectionCount);
+        settings.put(OTelLogsSourceConfig.RETRY_INFO, new RetryInfoConfig(Duration.ofMillis(50), Duration.ofMillis(100)));
```
[the remark this answers is not in the document]
Comment (i-vim) [reply data-prepper-plugins/otel-logs-source/src/test/java/org/opensearch/dataprepper/plugins/source/otellogs/OtelLogsSourceConfigTests.java:326]: done

File: data-prepper-plugins/otel-metrics-source/README.md (hunk 1 above)
Comment (i-vim) [reply data-prepper-plugins/otel-metrics-source/README.md:31]: done

File: data-prepper-plugins/otel-metrics-source/src/main/java/org/opensearch/dataprepper/plugins/source/otelmetrics/OTelMetricsSource.java (hunk 2 above)
Comment (i-vim) [reply data-prepper-plugins/otel-metrics-source/src/main/java/org/opensearch/dataprepper/plugins/source/otelmetrics/OTelMetricsSource.java:230]: done

File: data-prepper-plugins/otel-metrics-source/src/test/java/org/opensearch/dataprepper/plugins/source/otelmetrics/OtelMetricsSourceConfigTests.java (hunk 3 above)
Comment (i-vim) [reply data-prepper-plugins/otel-metrics-source/src/test/java/org/opensearch/dataprepper/plugins/source/otelmetrics/OtelMetricsSourceConfigTests.java:323]: done

File: data-prepper-plugins/otel-trace-source/README.md (hunk 4 above)
Comment (i-vim) [reply data-prepper-plugins/otel-trace-source/README.md:48]: done

File: data-prepper-plugins/otel-trace-source/src/main/java/org/opensearch/dataprepper/plugins/source/oteltrace/OTelTraceSource.java (hunk 5 above)
Comment (i-vim) [reply data-prepper-plugins/otel-trace-source/src/main/java/org/opensearch/dataprepper/plugins/source/oteltrace/OTelTraceSource.java:213]: done

**Review by KarstenSchnitter — APPROVED** [review]

**Pull request merged**